contracts/truefi2/interface/ILoanFactory2.sol

```diff
@@ -21,7 +22,7 @@ interface ILoanFactory2 {
 
     function isCreatedByFactory(IDebtToken) external view returns (bool);
 
-    function isLoanToken(IDebtToken) external view returns (bool);
+    function isLoanToken(ILoanToken2) external view returns (bool);
```

Comment:
Let's rename this to `isLoanToken(ILoanToken3)` and add a new function `isLegacyLoanToken(ILegacyLoanToken2)`